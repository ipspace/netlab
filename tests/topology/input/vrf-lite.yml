defaults.device: eos
provider: clab
  
module: [ vrf ]
defaults.Output: ["yaml=-"]

vrf.as: 65000
vrfs:
  red:
  blue:

nodes:
  r1:
  r2:
    vrfs:
      green:
  r3:
    vrfs:
      yellow:
        id: 1
      orange:
        rd: "65101:2"
        loopback:
      black:
        import: "65303:3"
        loopback:
          ipv4: 10.0.0.1

links:
- r1:
    vrf: red
  r2:
- r1:
    vrf: blue
  r3:
- r3:
    vrf: yellow
