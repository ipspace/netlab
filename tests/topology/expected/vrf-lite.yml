input:
- topology/input/vrf-lite.yml
- package:topology-defaults.yml
links:
- interfaces:
  - ipv4: 10.1.0.1/30
    node: r1
    vrf: red
  - ipv4: 10.1.0.2/30
    node: r2
  left:
    ifname: Ethernet1
    ipv4: 10.1.0.1/30
    node: r1
  linkindex: 1
  name: r1 - r2
  node_count: 2
  prefix:
    ipv4: 10.1.0.0/30
  right:
    ifname: Ethernet1
    ipv4: 10.1.0.2/30
    node: r2
  type: p2p
- interfaces:
  - ipv4: 10.1.0.5/30
    node: r1
    vrf: blue
  - ipv4: 10.1.0.6/30
    node: r3
  left:
    ifname: Ethernet2
    ipv4: 10.1.0.5/30
    node: r1
  linkindex: 2
  name: r1 - r3
  node_count: 2
  prefix:
    ipv4: 10.1.0.4/30
  right:
    ifname: Ethernet1
    ipv4: 10.1.0.6/30
    node: r3
  type: p2p
- bridge: input_3
  interfaces:
  - ipv4: 172.16.0.3/24
    node: r3
    vrf: yellow
  linkindex: 3
  node_count: 1
  prefix:
    ipv4: 172.16.0.0/24
  type: stub
module:
- vrf
name: input
nodes:
  r1:
    af:
      ipv4: true
      vpnv4: true
    box: ceos:4.26.4M
    clab:
      env:
        INTFTYPE: et
      kind: ceos
    device: eos
    hostname: clab-input-r1
    id: 1
    interfaces:
    - clab:
        name: et1
      ifindex: 1
      ifname: Ethernet1
      ipv4: 10.1.0.1/30
      linkindex: 1
      name: r1 -> r2
      neighbors:
      - ifname: Ethernet1
        ipv4: 10.1.0.2/30
        node: r2
      remote_id: 2
      remote_ifindex: 1
      type: p2p
      vrf: red
    - clab:
        name: et2
      ifindex: 2
      ifname: Ethernet2
      ipv4: 10.1.0.5/30
      linkindex: 2
      name: r1 -> r3
      neighbors:
      - ifname: Ethernet1
        ipv4: 10.1.0.6/30
        node: r3
      remote_id: 3
      remote_ifindex: 1
      type: p2p
      vrf: blue
    loopback:
      ipv4: 10.0.0.1/32
    mgmt:
      ifname: Management0
      ipv4: 192.168.121.101
      mac: 08-4F-A9-00-00-01
    module:
    - vrf
    name: r1
    vrf:
      as: 65000
    vrfs:
      blue:
        af:
          ipv4: true
        export: '65000:3'
        id: 3
        import: '65000:3'
        rd: '65000:3'
      red:
        af:
          ipv4: true
        export: '65000:2'
        id: 2
        import: '65000:2'
        rd: '65000:2'
  r2:
    af:
      ipv4: true
    box: ceos:4.26.4M
    clab:
      env:
        INTFTYPE: et
      kind: ceos
    device: eos
    hostname: clab-input-r2
    id: 2
    interfaces:
    - clab:
        name: et1
      ifindex: 1
      ifname: Ethernet1
      ipv4: 10.1.0.2/30
      linkindex: 1
      name: r2 -> r1
      neighbors:
      - ifname: Ethernet1
        ipv4: 10.1.0.1/30
        node: r1
      remote_id: 1
      remote_ifindex: 1
      type: p2p
    loopback:
      ipv4: 10.0.0.2/32
    mgmt:
      ifname: Management0
      ipv4: 192.168.121.102
      mac: 08-4F-A9-00-00-02
    module: []
    name: r2
    vrf:
      as: 65000
    vrfs:
      green:
        export: '65000:4'
        id: 4
        import: '65000:4'
        rd: '65000:4'
  r3:
    af:
      ipv4: true
      vpnv4: true
    box: ceos:4.26.4M
    clab:
      env:
        INTFTYPE: et
      kind: ceos
    device: eos
    hostname: clab-input-r3
    id: 3
    interfaces:
    - clab:
        name: et1
      ifindex: 1
      ifname: Ethernet1
      ipv4: 10.1.0.6/30
      linkindex: 2
      name: r3 -> r1
      neighbors:
      - ifname: Ethernet2
        ipv4: 10.1.0.5/30
        node: r1
      remote_id: 1
      remote_ifindex: 2
      type: p2p
    - bridge: input_3
      clab:
        name: et2
      ifindex: 2
      ifname: Ethernet2
      ipv4: 172.16.0.3/24
      linkindex: 3
      neighbors: []
      type: stub
      vrf: yellow
    loopback:
      ipv4: 10.0.0.3/32
    mgmt:
      ifname: Management0
      ipv4: 192.168.121.103
      mac: 08-4F-A9-00-00-03
    module:
    - vrf
    name: r3
    vrf:
      as: 65000
    vrfs:
      black:
        export: '65303:3'
        id: 3
        import: '65303:3'
        loopback:
          ipv4: 10.0.0.1
        rd: '65303:3'
      orange:
        export: '65101:2'
        id: 2
        import: '65101:2'
        loopback: null
        rd: '65101:2'
      yellow:
        af:
          ipv4: true
        export: '65000:1'
        id: 1
        import: '65000:1'
        rd: '65000:1'
provider: clab
vrf:
  as: 65000
vrfs:
  blue:
    export: '65000:3'
    id: 3
    import: '65000:3'
    rd: '65000:3'
  red:
    export: '65000:2'
    id: 2
    import: '65000:2'
    rd: '65000:2'
